### PR TITLE
144 sheltered animal stats

### PR DESCRIPTION
**Pull request by mindra**
Solving issue https://github.com/pets-oss/pets-back/issues/144
Would be nice to get  feedback

### CODE REVIEW

File: database/4-reports.sql [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+SELECT
+	to_char(aega.date_time, 'yyyymm') AS "year_month",
+	ad.breed_id,
+	a.organization,
+	'given away' AS "circumstance",
+	count(aega.animal_id) AS "cnt"
+FROM animal_event_given_away aega
+INNER JOIN animal_details ad ON ad.animal_id = aega.animal_id
+INNER JOIN animal a ON a.id = ad.animal_id
+GROUP BY to_char(aega.date_time, 'yyyymm'), ad.breed_id, a.organization
```
Comment (AzisK) [inline database/4-reports.sql:23]: Good job on even grouping by organization. Pardon me for not specifying this in the ticket. For now we can simply return the sheltered statistics of all organizations and later we can always plug in a WHERE filter to filter by organization id

File: database/4-reports.sql [hunk 2]
```diff
@@ -0,0 +1,36 @@
+-- https://github.com/pets-oss/pets-back/issues/144
+WITH cte_stats AS (
+SELECT
+	to_char(aef.date_time, 'yyyymm') AS "year_month",
+	ad.breed_id,
+	a.organization,
+	'found' AS "circumstance",
+	count(aef.animal_id) AS "cnt"
+FROM animal_event_found aef
+INNER JOIN animal_details ad ON ad.animal_id = aef.animal_id
```
Comment (AzisK) [inline database/4-reports.sql:10]: There is no need to write `INNER JOIN` as `JOIN` is defaulted to `INNER JOIN`

File: database/4-reports.sql (hunk 1 above)
Comment (AzisK) [reply database/4-reports.sql:23]: According to this ticket, we should group by species, not breed. Thus, I believe grouping by breed_id is not totally correct

File: database/4-reports.sql [hunk 3]
```diff
@@ -0,0 +1,36 @@
...
+)
+SELECT
+	rs.year_month,
+	st."translation" AS "species",
+	rs.circumstance,
+	o.Name AS "Organization",
+	rs.cnt
+FROM cte_stats rs
+INNER JOIN breed b ON b.id = rs.breed_id
+INNER JOIN species_translation st ON st."species" = b."species"
```
Comment (AzisK) [inline database/4-reports.sql:33]: As we have translations, we should specify the "lt" language here as well. If not and if we had more translations, we would get all of the translations

**Review by AzisK — CHANGES_REQUESTED** [review]
That's a good start. I left some comments for improvements.

This is a bit shorter solution that I propose (it's possible to come up with an even shorter one)

```sql
SELECT 
	EXTRACT(YEAR FROM aef.date_time) * 100 + EXTRACT(MONTH FROM aef.date_time) AS year_month,
	st."translation", 
	'found' AS circumstance,
	COUNT(*) cnt
FROM animal_event_found aef 
JOIN animal_details ad
    ON aef.animal_id = ad.animal_id 
JOIN breed b 
    ON ad.breed_id = b.id 
JOIN (SELECT * FROM species_translation WHERE language = 'lt') st 
    ON b."species" = st."species" 
GROUP by st."translation", year_month
UNION ALL
SELECT 
    EXTRACT(YEAR FROM agw.date_time) * 100 + EXTRACT(MONTH FROM agw.date_time) AS year_month,
	st."translation", 
	'given away' AS circumstance,
	COUNT(*) cnt
FROM animal_event_given_away agw 
JOIN animal_details ad
    ON agw.animal_id = ad.animal_id 
JOIN breed b 
    ON ad.breed_id = b.id 
JOIN (SELECT * FROM species_translation WHERE language = 'lt') st  
    ON b."species" = st."species" 
GROUP BY st."translation", year_month
ORDER BY year_month;
```

**Comment by AzisK** [comment]
@mindra are you going to update the code?

**Comment by mindra** [comment]
I'll do my best to push updates later today

File: database/4-reports.sql (hunk 2 above)
Comment (mindra) [reply database/4-reports.sql]: Updated

File: database/4-reports.sql (hunk 3 above)
Comment (mindra) [reply database/4-reports.sql]: Updated

File: database/4-reports.sql (hunk 1 above)
Comment (mindra) [reply database/4-reports.sql]: Updated

**Review by AzisK — APPROVED** [review]
LGTM

**Pull request merged**